lib/spack/spack/installer.py

```diff
@@ -648,6 +649,46 @@ def set(self, text):
         sys.stdout.flush()
 
 
+class TermStatusLine(object):
+    """
+    This class is used in distributed builds to inform the user other packages are
+    being installed by another process.
+    """
+    def __init__(self, enabled):
+        self.enabled = enabled
+        self.pkg_set = set()
+        self.pkg_list = []
+
+    def add(self, name):
```

Comment:
Can we use something else (the hash maybe?) to unify on specs being installed? We'll soon move to a model where e.g. we can have more that one `cmake` or `bazel` in the same spec.

Reply:
Same happens already in environments with `concretize: separately`.

`installer.py` internally uses `"{0}-{1}-{2}".format(pkg.name, pkg.version, pkg.spec.dag_hash())` as human readable unique identifier. But personally I think

```
Waiting for zlib-1.2.11-f55qopc3ymef6r5dnpod73iotnhpiaxo, zlib-1.2.11-jaetzxym7nrstpwturv3ujm45xfnl4tr, openmpi-4.1.2-3gf27zhfugk3qsyxpxpina7eksgekjc6...
```

doesn't look great. Maybe on separate lines it could work:

```
==> Waiting for zlib-1.2.11-f55qopc3ymef6r5dnpod73iotnhpiaxo...
==> Waiting for zlib-1.2.11-jaetzxym7nrstpwturv3ujm45xfnl4tr...
==> Waiting for openmpi-4.1.2-3gf27zhfugk3qsyxpxpina7eksgekjc6...
```

which is similar to what is already used:

```
==> Installing spla-1.5.2-ol6j2qqi62voj2sxgnmezby5nwz2bxsy
```

Reply:
Not sure why we actually print this odd triplet and not `spec.cshort_spec` :thinking: but I guess consistency with the rest of the installer is better

Reply:
I'd be fine with using `spec.format('{name}{/hash:7}') Would that be fine?